x-pack/test_serverless/api_integration/test_suites/common/platform_security/authorization.ts

```diff
@@ -76,5 +90,39 @@ export default function ({ getService }: FtrProviderContext) {
         });
       });
     });
+
+    describe('available features', () => {
+      const svlUserManager = getService('svlUserManager');
+      const supertestWithoutAuth = getService('supertestWithoutAuth');
+      let adminCredentials: { Cookie: string };
+
+      before(async () => {
+        // get auth header for Viewer role
+        adminCredentials = await svlUserManager.getApiCredentialsForRole('admin');
+      });
+
+      it('all Dashboard and Discover sub-feature privileges are disabled', async () => {
```

Comment:
__note:__ I'm still thinking what other tests would be beneficial to add, and happy to take any suggestions! The first thing that comes to my mind is to have a test per solution/project-type that validates `composedOf` for the security/dashboard/discover using `/privileges` API.

Reply:
I agree. We have similar tests for on-prem in `x-pack/test/api_integration/apis/security/privileges.ts` that you can use as a starting point.

I'd be interested to see if we could extend this to also assert on the actions within each of these privileges. For this, we would need an easy way to update the tests for future changes and contributors from outside our team. I could be mis-remembering, but I think we have support for proper snapshots in these FTR tests now?